### PR TITLE
created coordinator

### PR DESCRIPTION
**Pull request by kosskin**
creater coordinator

https://user-images.githubusercontent.com/80413929/209783388-9ac87c12-28a7-476d-b829-e00543c59f5c.mov

### CODE REVIEW

File: Coordinator4/Coordinator3/LoginVC/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+//  Created by Валентин Коскин on 27.12.2022.
+//
+
+import UIKit
+
+/// Authentivication screen
+final class LoginViewController: UIViewController {
+    // MARK: - Visual Components
+    
+    @IBOutlet weak var loginTextField: UITextField!
```
Comment (BurnashovD) [inline Coordinator4/Coordinator3/LoginVC/LoginViewController.swift:14]:
```suggestion
    @IBOutlet weak private var loginTextField: UITextField!
```
и ниже

File: Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//  Created by Валентин Коскин on 27.12.2022.
+//
+
+import UIKit
+
+/// Recovery password screen
+final class RecoveryPasswordViewController: UIViewController {
+    // MARK: - Visual Components
+
+    @IBOutlet weak var loginTextField: UITextField!
```
Comment (BurnashovD) [inline Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift:14]:
```suggestion
    @IBOutlet weak private var loginTextField: UITextField!
```

File: Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            return
+        }
+        showPassword()
+    }
+    
+    @IBAction private func goToAuth(_ sender: Any) {
+        navigationController?.popViewController(animated: false)
+    }
+    
+    private func showPassword() {
```
Comment (BurnashovD) [inline Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift:31]:
```suggestion
    private func showPasswordAlertAction() {
```

File: Coordinator4/Coordinator3/MainVC/MainViewController.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+//
+
+import UIKit
+
+/// Main screen
+final class MainViewController: UIViewController {
+    
+    // MARK: - Public Properties
+    
+    var onLogout: (() -> Void)?
```
Comment (death654321) [inline Coordinator4/Coordinator3/MainVC/MainViewController.swift:15]:
```suggestion
    var onLogoutHandler: (() -> Void)?
```

File: Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+//  Coordinator3
+//
+//  Created by Валентин Коскин on 27.12.2022.
+//
+
+import UIKit
+
+/// Recovery password screen
+final class RecoveryPasswordViewController: UIViewController {
+    // MARK: - Visual Components
```
Comment (death654321) [inline Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift:12]:
```suggestion
    // MARK: - Outlets
```

File: Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    @IBOutlet weak private var loginTextField: UITextField!
+    
+    // MARK: - Private Methods
+    
+    @IBAction private func recovery(_ sender: Any) {
```
Comment (death654321) [inline Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift:18]:
```suggestion
    @IBAction private func recoveryAction(_ sender: Any) {
```

File: Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        guard
+            let login = loginTextField.text,
+            login == Constants.login else {
+            return
+        }
+        showPasswordAlertAction()
+    }
+    
+    @IBAction private func goToAuth(_ sender: Any) {
```
Comment (death654321) [inline Coordinator4/Coordinator3/RecoveryPasswordVC/RecoveryPasswordViewController.swift:27]:
```suggestion
    @IBAction private func goToAuthAction(_ sender: Any) {
```

**Review by death654321 — APPROVED** [review]

**Pull request merged**